pkg/operator/target_config_reconciler.go

```diff
@@ -123,47 +122,55 @@ func (c TargetConfigReconciler) sync() error {
 }
 
 func (c *TargetConfigReconciler) manageConfigMap(secondaryScheduler *secondaryschedulersv1.SecondaryScheduler) (*v1.ConfigMap, bool, error) {
-	required := resourceread.ReadConfigMapV1OrDie(bindata.MustAsset("assets/secondary-scheduler/configmap.yaml"))
-	required.Name = secondaryScheduler.Name
-	required.Namespace = secondaryScheduler.Namespace
-	required.OwnerReferences = []metav1.OwnerReference{
-		{
-			APIVersion: "v1",
-			Kind:       "SecondaryScheduler",
-			Name:       secondaryScheduler.Name,
-			UID:        secondaryScheduler.UID,
-		},
-	}
+	var required *v1.ConfigMap
+	var yamlFile []byte
+	var err error
 
 	if secondaryScheduler.Spec.SchedulerConfig == secondaryschedulersv1.CustomizedConfig {
-		yamlFile, err := ioutil.ReadFile(SecondarySchedulerCommand[2])
-		if err != nil {
-			return nil, false, err
-		}
-		required.Data = map[string]string{"config.yaml": string(yamlFile)}
-	} else {
-		yamlFile := bindata.MustAsset("assets/schedulerconfig/" + string(secondaryScheduler.Spec.SchedulerConfig) + ".yaml")
+		required, err = c.kubeClient.CoreV1().ConfigMaps(secondaryScheduler.Namespace).Get(context.TODO(), "customized", metav1.GetOptions{})
 
-		// Replace the ${PROM_URL} and ${PROM_TOKEN}
-		promAddress, promToken, err := getPromInfo(c.kubeClient, c.osrClient)
 		if err != nil {
-			fmt.Println(err)
+			klog.Errorf("Cannot find ConfigMap %s for the secondaryscheduler", secondaryschedulersv1.CustomizedConfig)
 			return nil, false, err
 		}
 
-		promInfo := map[string]string{
-			"${PROM_URL}":   promAddress,
-			"${PROM_TOKEN}": promToken,
+		klog.Infof("Find ConfigMap %s for the secondaryscheduler.", secondaryScheduler.Spec.SchedulerConfig)
+		yamlFile = []byte(required.Data["config.yaml"])
+	} else {
+		klog.Infof("Load SchedulerConfig %s as the configmap for the secondary-scheduler.", secondaryScheduler.Spec.SchedulerConfig)
+		required = resourceread.ReadConfigMapV1OrDie(bindata.MustAsset("assets/secondary-scheduler/configmap.yaml"))
+		required.Name = secondaryScheduler.Name
+		required.Namespace = secondaryScheduler.Namespace
+		required.OwnerReferences = []metav1.OwnerReference{
+			{
+				APIVersion: "v1",
+				Kind:       "SecondaryScheduler",
+				Name:       secondaryScheduler.Name,
+				UID:        secondaryScheduler.UID,
+			},
 		}
+		yamlFile = bindata.MustAsset("assets/schedulerconfig/" + string(secondaryScheduler.Spec.SchedulerConfig) + ".yaml")
+	}
 
-		for promKey, promVal := range promInfo {
-			tmp := bytes.Replace(yamlFile, []byte(promKey), []byte(promVal), -1)
-			yamlFile = tmp
-		}
+	// Replace the ${PROM_URL} and ${PROM_TOKEN}
+	promAddress, promToken, err := getPromInfo(c.kubeClient, c.osrClient)
```

Comment:
Hi, @damemi , yes. There are other ways to get the route and the token.
To get the route of Prometheus, we can use 
```
oc get routes prometheus-k8s -n openshift-monitoring -ojson |jq ".status.ingress"|jq ".[0].host"|sed 's/"//g'
```

To get the available token names for the Prometheus, we can run the following and export the secret name to `TOKEN_NAME`
```
oc get secret -n openshift-monitoring|awk '{print $1}'
```

To get the actual Prometheus token, we can run
```
oc describe secret $TOKEN_NAME -n openshift-monitoring|grep "token:"|cut -d: -f2|sed 's/^ *//g'
```

If users prefer to set up their own token, our controllers will not kick in as there would be no `${PROM_URL}` and `${PROM_TOKEN}` to be replaced.

Reply:
Ok, and these are used for the Trimaran plugins to access load-aware metrics correct? In that case then for the same reason that we do not include support for the Trimaran plugins by default I don't think we should have code in the operator that is intended to work with those plugins.

For the first iteration of this operator, it should only be the most basic and generic functions of deploying a scheduler. Our intent right now is to only automate the steps of deploying a scheduler and passing a config to it. So, we should remove these functions for now too.

If we do start officially supporting Trimaran with this operator, automated parsing like this will be a great helpful addition. For now however, we only want the basics and a clean codebase to build off of.

Reply:
@damemi , sure, I can remove those pieces of code and only add them back when we want to support it.